src/com/ableton/VirtualEnv.groovy

```diff
@@ -0,0 +1,30 @@
+package com.ableton
+
+
+class VirtualEnv implements Serializable {
+  def script
+  def destDir
+
+  VirtualEnv(script, python) {
+    this.script = script
+    def pathSeparator = script.isUnix() ? "/" : "\\"
+    this.destDir = script.pwd(tmp:true) +
+      pathSeparator +
+      script.env.BUILD_NUMBER +
+      pathSeparator +
+      python
+  }
+
+
+  def installRequirements() {
+    run("pip install -r requirements.txt")
+  }
+
+
+  def run(command) {
+    script.sh("""
+      . ${destDir}/bin/activate
```

Comment:
Can we use `source` instead? How does it work on Windows?

Reply:
Nope, `source` is a internal shell command but is technically an alias for `.` (which I learned when googling for how to source a script in Jenkins). So this is the correct way to do this.

It will work on Windows assuming that `sh` works at all on Windows. For the record, there is also a `bat` call for that platform.

Reply:
Sounds good.